llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp

```diff
@@ -4378,7 +4378,8 @@ void CodeGenDAGPatterns::ParsePatterns() {
       FindPatternInputsAndOutputs(Pattern, Pattern.getTree(j), InstInputs,
                                   InstResults, InstImpResults);
 
-    ParseOnePattern(CurPattern, Pattern, Result, InstImpResults);
+    ParseOnePattern(CurPattern, Pattern, Result, InstImpResults,
```

Comment:
Can you just skip everything here if it's ignored? This currently defers it to as late as possible, and still forces you through all the rest of the machinery

Reply:
It looks like this is possible. Trying it out.

Reply:
Disabling both is not useful though. The only reason is to remove some globalisel migration pain

Reply:
Trying to name it after its functionality, not necessarily its usage. We originally thought it to disable imports in the SelectionDAG, but it disables the imports from the perspective of GISel; so calling it with a GI prefix makes sense, but it has more general usage than that.